usaspending_api/search/v2/views/spending_over_time.py

```diff
@@ -89,31 +102,107 @@ def database_data_layer(self):
 
         return queryset, values
 
+    def apply_elasticsearch_aggregations(self, search: Search) -> None:
+        """
+        Takes in an instance of the elasticsearch-dsl.Search object and applies the necessary
+        aggregations in a specific order to get expected results.
+        """
+        interval = "year" if self.group == "fiscal_year" else self.group
+
+        # The individual aggregations that are needed; with two different sum aggregations to handle issues with
+        # summing together floats.
+        group_by_time_period_agg = A(
+            "date_histogram", field="fiscal_action_date", interval=interval, format="yyyy-MM-dd"
+        )
+        sum_as_cents_agg = A(
+            "sum", script={"lang": "painless", "source": f"doc['generated_pragmatic_obligation'].value * 100"}
+        )
+        sum_as_dollars_agg = A(
+            "bucket_script", buckets_path={"sum_as_cents": "sum_as_cents"}, script="params.sum_as_cents / 100"
+        )
+
+        # Putting the aggregations together; in order for the aggregations to the correct structure they
+        # unfortunately need to be one after the other. This allows for nested aggregations as opposed to sibling.
+        search.aggs.bucket("group_by_time_period", group_by_time_period_agg).metric(
+            "sum_as_cents", sum_as_cents_agg
+        ).pipeline("sum_as_dollars", sum_as_dollars_agg)
+
+    def parse_elasticsearch_bucket(self, bucket):
+        key_as_date = pd.to_datetime(bucket["key_as_string"], format="%Y-%m-%d")
```

Comment:
Does pandas have an advantage over datetime.strptime()?

Reply:
Pandas datetime objects have a `quarter` property unlike normal datetime objects. Could implement without pandas, but since it is already in the project I figured we should use that instead of datetime.

Reply:
Makes sense. Although I would like to remove pandas from our app. It's a large package with lots of dependencies.

Reply:
Sad to see it go, but understandable. I will go ahead and make the change.

Reply:
Up to you, we aren't ready to pull the plug yet.